modules/context/repo.go

```diff
@@ -100,6 +102,23 @@ func (r *Repository) CanUseTimetracker(issue *models.Issue, user *models.User) b
 		r.IsWriter() || issue.IsPoster(user.ID) || issue.AssigneeID == user.ID)
 }
 
+// GetCommitsCount returns cached commit count for current view
+func (r *Repository) GetCommitsCount() (int64, error) {
+	if r.IsViewBranch {
+		return cache.GetInt64(fmt.Sprintf("branch-commits-count-%d-%s", r.Repository.ID, r.BranchName), func() (int64, error) {
```

Comment:
To me it feels like this should be in the model instead. Because now we end up with some places having it cached, and other places direct access 😕

Reply:
Theoretically I can move it to model and make it less context dependant as branch and tag names can not be the with same name anyway